corpdir-srv-sync/src/main/java/com/github/bordertech/corpdir/sync/api/mapper/AbstractVersionMapper.java

```diff
@@ -0,0 +1,21 @@
+package com.github.bordertech.corpdir.sync.api.mapper;
+
+import com.github.bordertech.corpdir.api.common.ApiKeyIdObject;
+import com.github.bordertech.corpdir.api.common.ApiVersionable;
+
+/**
+ * Abstract mapper to copy API versioned object common fields.
+ * @author aswinkandula
+ * @param <A>
+ */
+public abstract class AbstractVersionMapper<A extends ApiKeyIdObject & ApiVersionable> extends AbstractMapper<A> {
```

Comment:
Mapper logic is looking good.

With `<A extends ApiKeyIdObject & ApiVersionable>` you can just do `<A extends ApiVersionable>` as `ApiVersionable` extends `ApiKeyIdObject`.